Product Collection: Support registering product collection that isn't displayed in Collection Chooser

imanish003 opened the pull request:
### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes #50762 

This PR enhances product collection registration by using `scope` argument for controlling visibility.

1. Collections with `scope: ['block', 'inserter']` (default):
   - Visible in the Collection Chooser
   - Accessible via the "Choose Collection" toolbar button

2. Collections with `scope: ['inserter']`:
   - Hidden from the Collection Chooser
   - "Choose Collection" toolbar button will not be visible
   - Visible in the block inserter

3. Collections with `scope: ['block']`:
   - Visible in the Collection Chooser
   - "Choose Collection" toolbar button will be visible
   - Not visible in the block inserter

4. Collections with `scope: []`:
   - Hidden from the Collection Chooser
   - "Choose Collection" toolbar button will not be visible
   - Not visible in the block inserter

In short, `block` scope means that the collection will be shown in Collection Chooser and "Choose Collection" toolbar button will be visible.

Related Slack discussion: p1724747359216099-slack-C05UMTA7W0P

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

**Prerequisite:** Download and install following plugin: [product-collection-testing.zip](https://github.com/user-attachments/files/17411745/product-collection-testing.zip). This plugin contains 3 custom collections named "My Custom Collection - with block scope", "My Custom Collection - with inserter scope", and "My Custom Collection - without scope argument".


<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

**Case 1: Collection has scope: ['block']**

1. Create a new post or page and add the Product Collection block.
2. Verify that in collection chooser, you can see collection with name "My Custom Collection - with block scope"
	- Sidenote: This collection has scope: ['block']
3. Click on "My Custom Collection - with block scope" to select it. 
4. Verify that "Choose Collection" toolbar button is visible

**Case 2: Collection has scope: ['inserter']**	

1. Create a new post or page and add the Product Collection block.
2. Verify that in collection chooser, you can't see collection with name "My Custom Collection - with inserter scope"
3. Verify that in inserter, you can see collection with name "My Custom Collection - with inserter scope"
4. Add "My Custom Collection - with inserter scope" to the post from inserter.
5. Verify that "Choose Collection" toolbar button is not visible

**Case 3: Collection has no scope argument**
_Sidenote_: When scope argument is not defined, default scope is ['block', 'inserter']

1. Create a new post or page and add the Product Collection block.
2. Verify that in collection chooser, you can see collection with name "My Custom Collection - without scope argument"
3. Verify that in inserter, you can see collection with name "My Custom Collection - without scope argument"
4. Add "My Custom Collection - without scope argument" to the post from inserter.
5. Verify that "Choose Collection" toolbar button is visible


<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [x] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [x] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

Product Collection: Support registering product collection that isn't displayed in Collection Chooser

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

imanish003 committed:
Filter product collection variations by 'block' scope

This commit updates the product collection feature to only show variations
with the 'block' scope in the collection chooser. To achieve this:

1. Added 'block' to the scope array for various product collection variations
   (best-sellers, featured, hand-picked, new arrivals, on-sale, related,
   top-rated, and upsells).
2. Modified the collection chooser logic to filter variations based on the
   'block' scope.

imanish003 committed:
Product Collection: Hide collection chooser toolbar for non-block scoped collections

This commit modifies the ToolbarControls component in the Product Collection block to conditionally render the CollectionChooserToolbar. The toolbar is now only displayed for collections with a scope that includes 'block'.

imanish003 committed:
Add scope property to product collections in E2E tests

This commit adds the 'scope' property to all registered product collections
in the E2E test plugin. The 'scope' is set to ['block'] for each collection.

Commit:
Add changefile(s) from automation for the following project(s): woocommerce-blocks

kmanijak commented on plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/toolbar-controls/index.tsx:
```diff
@@ -24,11 +25,19 @@ export default function ToolbarControls(
 	);
 
+	const collection = getCollectionByName( props.attributes.collection );
+	const isShowCollectionChooserToolbar =
```
Sorry, I don't understand this variable name. Did you mean `showCollectionChooserToolbar`?

imanish003 replied:
Yes, Made the changes in [02ca2de](https://github.com/woocommerce/woocommerce/pull/52057/commits/02ca2de245e8f85551716d5dc642e75bd7afd1d5)

kmanijak reviewed (commented):
I left one minor comment regarding variable name. One thing I'd request is to update documentation regarding `scope` variable. Because we're aligned with Variations API but it's a custom "interpretation" or "implementation" of `scope: [ 'block' ]` (it occurs a bit differently in Variations).

imanish003 committed:
Rename isShowCollectionChooserToolbar to showCollectionChooserToolbar
for better readability.

imanish003 committed:
Add documentation and refactor for product collection scope

This commit introduces the following changes:

1. Update documentation to explain the `scope` argument for product collections
2. Refactor Collection Chooser and toolbar controls to handle default scope
3. Remove explicit `scope` from test collections to align with default behavior

imanish003 committed:
Add E2E tests for Product Collection scope argument

This commit adds E2E tests to verify the behavior of Product Collections
registered with different scopes. It includes:
- Tests for collections with 'inserter' and 'block' scopes
- Verification of visibility in Collection Chooser
- Checks for 'Choose collection' button visibility

imanish003 committed:
Merge branch 'trunk' of https://github.com/woocommerce/woocommerce into add/50762-product-collection-support-registering-product-collection-that-isnt-displayed-in-inserter

imanish003 committed:
Generate manifest file

imanish003 commented:
Hey @kmanijak, This PR is ready for review again:
- Added E2E tests
- Updated documentation

imanish003 committed:
Generate manifest file

imanish003 committed:
docs: Remove unsupported emojis from product collection documentation

kmanijak reviewed (approved):
All looks good and tests well! Thanks for adding E2E tests as well! Great work!

Could you just update the PR description a bit? Currently it says:

> 1. Collections with scope: ['block', 'inserter'] (default):
> ...
> 2. Collections with scope: ['inserter']:
> ...
> 3. Collections with empty scope or without 'block':

2 and 3 overlap. I think we should explicitly describe remaining cases (`['block]` and `[]`):

1. Collections with scope: ['block', 'inserter'] (default)
2. Collections with scope: ['inserter']
3. Collections with scope: ['block']
4. Collections with scope: []

imanish003 committed:
Merge branch 'trunk' into add/50762-product-collection-support-registering-product-collection-that-isnt-displayed-in-inserter

imanish003 commented:
> 2 and 3 overlap. I think we should explicitly describe remaining cases (`['block]` and `[]`):
> 
> 1. Collections with scope: ['block', 'inserter'] (default)
> 2. Collections with scope: ['inserter']
> 3. Collections with scope: ['block']
> 4. Collections with scope: []

@kmanijak Good point! I have made the changes to PR description 🤝

xristos3490 commented:
Thanks for working on this, @imanish003! Now that it's merged, I'll initiate the work to update our premium offerings in the Product Recommendations extension!
